pkgs/development/libraries/linenoise/default.nix

```
@@ -1,29 +1,36 @@
-{ lib, stdenv, fetchFromGitHub }:
+{ fetchFromGitHub, lib, stdenv, validatePkgConfig }:
 
 stdenv.mkDerivation {
   pname = "linenoise";
-  version = "1.0.10";  # Its version 1.0 plus 10 commits
+  version = "1.0-34-g93b2db9";
 
   src = fetchFromGitHub {
     owner = "antirez";
     repo = "linenoise";
-    rev = "c894b9e59f02203dbe4e2be657572cf88c4230c3";
-    sha256 = "0wasql7ph5g473zxhc2z47z3pjp42q0dsn4gpijwzbxawid71b4w";
+    rev = "1.0-34-g93b2db9";
+    hash = "sha256-GsrYg16gpjHkkmpCU3yGzqNS/buZl+JoWALLvwzmT4A=";
   };
 
-  buildPhase = ./create-pkg-config-file.sh;
+  nativeBuildInputs = [ validatePkgConfig ];
+
+  buildPhase = ''
+    $CC -c -gctf -gdwarf -o linenoise.o linenoise.c
+    $CC -shared -o liblinenoise.so linenoise.o
+    $AR rcs liblinenoise.a linenoise.o
+  '';
```

```suggestion
  buildPhase = ''
    runHook preBuild

    $CC -c -gctf -gdwarf -o linenoise.o linenoise.c
    $CC -shared -o liblinenoise.so linenoise.o
    $AR rcs liblinenoise.a linenoise.o

    runHook postBuild
  '';
```
please add our hooks